pyrsmi/rocml.py

```
@@ -736,6 +984,25 @@ def smi_get_device_id(dev):
     """
     try:
         handle = _get_processor_handle(dev)
+        
+        # Use amdsmi package if available
+        if _using_amdsmi_package:
+            try:
+                import amdsmi
+                asic_info = amdsmi.amdsmi_get_gpu_asic_info(handle)
+                device_id = asic_info.get('device_id', -1)
+                # amdsmi package returns device_id as hex string like '0x75a0'
+                if isinstance(device_id, str):
+                    try:
+                        return int(device_id, 16) if device_id.startswith('0x') else int(device_id)
+                    except ValueError:
+                        return -1
```

I'd have thought that a device_id not matching a recognized format is an error to raise, rather than just swap for -1 as a default, because it might indicate that a whole new class of devices isn't handled here. Is it better to raise the error here?